Rubberduck.Parsing/ComReflection/ComMember.cs

```diff
@@ -100,13 +103,24 @@ private void LoadParameters(FUNCDESC funcDesc, ITypeInfo info)
             var names = new string[255];
             info.GetNames(Index, names, names.Length, out int count);
 
-            for (var index = 0; index < count - 1; index++)
+            for (var index = 0; index < funcDesc.cParams; index++)
             {
                 var paramPtr = new IntPtr(funcDesc.lprgelemdescParam.ToInt64() + Marshal.SizeOf(typeof(ELEMDESC)) * index);
                 var elemDesc = Marshal.PtrToStructure<ELEMDESC>(paramPtr);
                 var param = new ComParameter(this, elemDesc, info, names[index + 1] ?? $"{index}unnamedParameter");
                 _parameters.Add(param);
             }
+
+            // See https://docs.microsoft.com/en-us/windows/desktop/midl/propput
+            // "A function that has the [propput] attribute must also have, as its last parameter, a parameter that has the [in] attribute."
+            if (funcDesc.invkind.HasFlag(INVOKEKIND.INVOKE_PROPERTYPUTREF) ||
+                funcDesc.invkind.HasFlag(INVOKEKIND.INVOKE_PROPERTYPUT))
+            {
+                AsTypeName = _parameters.Last();
+                _parameters = _parameters.Take(funcDesc.cParams - 1).ToList();
```

Comment:
I wonder if we really want to use LINQ. As a rule we avoid using LINQ with COM objects, and type lib stuff are COM objects. Are we inadvertently creating leaks this way?

Reply:
This should be perfectly safe in this context. `ComParameter` is a completely managed type.